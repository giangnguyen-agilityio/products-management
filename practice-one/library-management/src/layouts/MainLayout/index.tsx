import React, {Suspense, useEffect, useRef, useState} from 'react'
import {Outlet, useNavigate} from 'react-router-dom'

// Importing the layouts
import Header from '@layouts/Header'
import LoadingPage from '@pages/LoadingPage'
import Footer from '@layouts/Footer'

// Importing the Book provider
import BookProvider from '@stores/books/BookProvider'

// Importing the helper functions
import {getItemInLocalStorage, clearLocalStorage} from '@helpers'

const MainLayout = (): JSX.Element => {
  const [isLogin, setIsLogin] = useState(false)
  const memberId: string = getItemInLocalStorage('memberId')
  const navigate = useNavigate()

  useEffect(() => {
    if (memberId) {
      setIsLogin(!!memberId)
      return
    }
    navigate('/login')
  }, [memberId])

  const handleSignIn = (): void => {
    navigate('/login')
  }

  const handleLogout = (): void => {
    clearLocalStorage()
    navigate('/login')
    setIsLogin(false)
  }
  return (
    <>
      <div className="wrapper">
        <div className="container">
          <Header
            isLogin={isLogin}
            onSignIn={handleSignIn}
            onLogout={handleLogout}
          />
          <main className="main-content">
            <BookProvider>
              <Suspense fallback={<LoadingPage />}>
                <Outlet />
              </Suspense>
            </BookProvider>
          </main>
          <Footer />
        </div>
      </div>
    </>
  )
}

export default MainLayout
