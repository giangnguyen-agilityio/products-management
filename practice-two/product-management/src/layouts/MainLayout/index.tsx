import { Suspense } from 'react'
import { Outlet } from 'react-router-dom'
import Header from '@layouts/Header'
import Footer from '@layouts/Footer'
import { Box } from '@chakra-ui/react'
import ProductProvider from '@stores/products/ProductProvider'
import Loading from '@components/common/Loading'
import { ErrorBoundary } from '@components/ErrorBoundary'
import Hero from '@components/common/Hero'
import { heroSectionContent } from '@constants'
import Contact from '@components/Contact'

const MainLayout = (): JSX.Element => {
  return (
    <ErrorBoundary>
      <Box padding={{ base: '22px 10px', md: '22px 50px' }} overflow="hidden">
        <Header />
        <Hero
          imageUrl={heroSectionContent.imageUrl}
          buttonHref={heroSectionContent.link}
          title={heroSectionContent.title}
          description={heroSectionContent.description}
        />
        <Suspense fallback={<Loading />}>
          <ProductProvider>
            <Outlet />
          </ProductProvider>
        </Suspense>
        <Contact />
        <Footer />
      </Box>
    </ErrorBoundary>
  )
}

export default MainLayout
