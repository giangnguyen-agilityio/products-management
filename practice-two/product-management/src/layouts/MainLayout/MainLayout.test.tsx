import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { ChakraProvider } from '@chakra-ui/react'
import MainLayout from '.'
describe('MainLayout component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <ChakraProvider>
        <MemoryRouter>
          <MainLayout />
        </MemoryRouter>
      </ChakraProvider>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
