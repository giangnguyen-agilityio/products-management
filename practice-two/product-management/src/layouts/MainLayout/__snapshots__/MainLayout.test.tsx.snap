// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-zixof1"
  >
    <header
      class="chakra-container header css-1mr40xt"
    >
      <div
        class="css-gg4vpm"
      >
        <a
          href="/"
        >
          <div
            class="logo css-759u60"
          />
        </a>
        <button
          aria-label="Open Menu"
          class="chakra-button hamburger-menu css-1lguxu9"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-onkibi"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M 3 5 A 1.0001 1.0001 0 1 0 3 7 L 21 7 A 1.0001 1.0001 0 1 0 21 5 L 3 5 z M 3 11 A 1.0001 1.0001 0 1 0 3 13 L 21 13 A 1.0001 1.0001 0 1 0 21 11 L 3 11 z M 3 17 A 1.0001 1.0001 0 1 0 3 19 L 21 19 A 1.0001 1.0001 0 1 0 21 17 L 3 17 z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </header>
    <footer
      class="chakra-container footer css-vhc7ah"
    >
      <div
        class="css-hv056j"
      >
        <ul
          class="footer-left css-j7qwjs"
        >
          <li
            class="css-k008qs"
          >
            <a
              href="/"
            >
              <div
                class="logo css-759u60"
              />
            </a>
          </li>
          <li
            class="css-k008qs"
          >
            <span
              class="chakra-text css-lsakg3"
            >
              Copyright Â© 2023, Giang Nguyen
              <br />
              All rights reserved.
            </span>
          </li>
          <li
            class="link-social css-1ov7iep"
          >
            <a
              href="/"
            >
              <div
                class="css-1wojifn"
              >
                <img
                  alt="The social icon"
                  class="chakra-image css-v67zxb"
                  src=""
                />
              </div>
            </a>
            <a
              href="/"
            >
              <div
                class="css-1wojifn"
              >
                <img
                  alt="The social icon"
                  class="chakra-image css-v67zxb"
                  src=""
                />
              </div>
            </a>
            <a
              href="/"
            >
              <div
                class="css-1wojifn"
              >
                <img
                  alt="The social icon"
                  class="chakra-image css-v67zxb"
                  src=""
                />
              </div>
            </a>
          </li>
        </ul>
        <ul
          class="footer-right css-1jnspk1"
        >
          <li
            class="footer-column css-j7qwjs"
          >
            <span
              class="chakra-text css-1c6t563"
            >
              Discover
            </span>
            <ul
              class="footer-column-items css-j7qwjs"
            >
              <li
                class="css-1l0pjoz"
              >
                <a
                  href="/"
                >
                  <span
                    class="chakra-text footer-link css-14cj6ry"
                  >
                    Home
                  </span>
                </a>
              </li>
              <li
                class="css-1l0pjoz"
              >
                <a
                  href="/"
                >
                  <span
                    class="chakra-text footer-link css-14cj6ry"
                  >
                    Products
                  </span>
                </a>
              </li>
            </ul>
          </li>
          <li
            class="footer-column css-j7qwjs"
          >
            <span
              class="chakra-text css-1c6t563"
            >
              About Us
            </span>
            <ul
              class="footer-column-items css-j7qwjs"
            >
              <li
                class="css-1l0pjoz"
              >
                <a
                  href="/"
                >
                  <span
                    class="chakra-text footer-link css-14cj6ry"
                  >
                    Privacy Policy
                  </span>
                </a>
              </li>
              <li
                class="css-1l0pjoz"
              >
                <a
                  href="/"
                >
                  <span
                    class="chakra-text footer-link css-14cj6ry"
                  >
                    Service Terms
                  </span>
                </a>
              </li>
              <li
                class="css-1l0pjoz"
              >
                <a
                  href="/"
                >
                  <span
                    class="chakra-text footer-link css-14cj6ry"
                  >
                    Contact
                  </span>
                </a>
              </li>
            </ul>
          </li>
          <li
            class="footer-column css-j7qwjs"
          >
            <span
              class="chakra-text css-1c6t563"
            >
              Help
            </span>
            <ul
              class="footer-column-items css-j7qwjs"
            >
              <li
                class="css-1l0pjoz"
              >
                <a
                  href="/"
                >
                  <span
                    class="chakra-text footer-link css-14cj6ry"
                  >
                    Help Center
                  </span>
                </a>
              </li>
              <li
                class="css-1l0pjoz"
              >
                <a
                  href="/"
                >
                  <span
                    class="chakra-text footer-link css-14cj6ry"
                  >
                    Report a Problem
                  </span>
                </a>
              </li>
            </ul>
          </li>
        </ul>
      </div>
    </footer>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
